### ISSUE TITLE
Value from return is evaluated an extra time

### ISSUE THREAD

**Issue by technoblogy** [technoblogy/ulisp-esp#79]
The value returned by return is incorrectly evaluated an extra time if the **return** clause is not the last expression within a **progn** and is within a **loop**, **dotimes**, or **dolist** block.

Expected:
````text
21808> (let ((a 7)) (loop (progn (return 'a))))
a
````
Observed:
````text
21808> (let ((a 7)) (loop (progn (return 'a) nil)))
7
````
Will be fixed next release.

**Comment by technoblogy** [comment]
My issue report is badly worded. the test to try is:

    (let ((a 7)) (loop (progn (return 'a) nil)))

Correct answer 'a', wrong answer '7'.

**Comment by technoblogy** [comment]
This is fixed in Version 4.6.
